src/dune_rules/pkg_rules.ml

```diff
@@ -673,21 +673,23 @@ module Action_expander = struct
     let dir = Path.build expander.paths.source_dir in
     match action with
     | Run (exe, args) ->
-      let* exe, more_args = Expander.expand_exe expander exe in
-      let+ args = Memo.parallel_map args ~f:(Expander.expand_sw expander) in
+      let+ exe, more_args = Expander.expand_exe expander exe
+      and+ args = Memo.parallel_map args ~f:(Expander.expand_sw expander) in
       let args = more_args @ List.concat args |> Value.L.to_strings ~dir in
       Action.Run (exe, Array.Immutable.of_list args)
     | Progn t ->
       let+ args = Memo.parallel_map t ~f:(expand ~expander) in
       Action.Progn args
     | System arg ->
-      let+ arg = Expander.expand_pform_gen ~mode:Single expander arg in
-      let arg = arg |> Value.to_string ~dir in
+      let+ arg =
+        Expander.expand_pform_gen ~mode:Single expander arg >>| Value.to_string ~dir
+      in
```

Comment:
Why is this better? It seems we now have two maps rather than just the one from before?

Reply:
You could also do that by inlining the arg but doesn't matter. I was simply interested if there was a deeper reason.